Realm/RLMSyncConfiguration.mm

```diff
@@ -140,11 +142,16 @@ - (instancetype)initWithUser:(RLMSyncUser *)user
             errorHandler = [=](std::shared_ptr<SyncSession> errored_session,
                                SyncError error) {
                 RLMSyncSession *session = [[RLMSyncSession alloc] initWithSyncSession:errored_session];
+                NSMutableDictionary *userInfo = [@{} mutableCopy];
```

Comment:
This is kinda needlessly clever (and since we know the size of the dictionary could be insignificantly faster as `[NSMutableDictionary dictionaryWithCapacity:error.user_info.size()]`).